actions/docs-verifier/src/RedirectionVerifier/DocfxConfiguration.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace RedirectionVerifier
+{
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes → JSON
+    internal sealed record DocfxConfiguration(
+        [property: JsonPropertyName("build")] DocfxBuild? Build)
+    {
+        private Matcher? _matcher;
+
+        private IEnumerable<string> GetExcludes()
+            // This method retrieves all the "exclude" arrays under each of the "content"s.
+            // This can have "false negatives", since we are excluding patterns that can be meant to apply only to cross-reference repositories.
+            // For example, https://github.com/dotnet/docs/blob/bc8e38479d9867c7ed4b308be596ee7642422754/docfx.json#L48-L69
+            // The exclude patterns in the link above are meant only for dotnet/csharplang repository. But we're taking them into account.
+            // If a file in the docs repo happen to match one of these patterns, it won't require a redirection, while it should.
+            // TODO: Fix that. One way to fix that is to add a flag to docfx.json for "content"s that are published.
+            // Another way is to not require any changes in docfx.json, but try to somehow find which "content" should be taken.
```

Comment:
No, let's not get all the excludes. Let's get only the one that is relevant to the repo in context. Use the `src` and match it on the repo name for the GitHub context object.

See https://github.com/dotnet/docs/blob/bc8e38479d9867c7ed4b308be596ee7642422754/docfx.json#L15, the src is `"docs"` -- only these things should be excluded. And if you cannot find any matching by src, there is nothing to exclude.

Reply:
@IEvangelist It's not always the case from what I see in AspNetCore.Docs repo.

https://github.com/dotnet/AspNetCore.Docs/blob/4de0bb40c02f0e00f2c1f377148ccf3d81b03627/aspnetcore/docfx.json#L21

Reply:
Either match the name, or `"."`. `"."` just means the root, that should work -- right?

Reply:
@IEvangelist Is there a guarantee that the "dest" should either be `.` or the repo name? I'm afraid if checking the repo name, e.g "docs" can cause problems in the private localized repositories that are probably mirrored from the original repo?

I think one good way to do that is to try treating "dest" as a relative directory and see if it exists or not. There should only be one match. What do you think?

Reply:
Do you mean "src"? The "dest" is the destination for static file gen, the src is the source directory to build from. What I'm suggesting is that if you're looking for `excludes`, you should only consider `excludes` that A) have a `src` that matches the repo name, or B) a `src` that is `.`

Reply:
Yeah I meant "src" instead of "dest" 😄 

>  A) have a src that matches the repo name

this is the part I'm afraid to cause issues. "src" might not match the repo name nor ".", it could be any folder name. and for localized repo (e.g, `dotnet/docs.<LanguageCode>`). The "src" would be "docs", not matching the repo name. That's why I'm thinking about validating files in disk.

Reply:
So, if you're to consider all the found exclude files you might end up excluding things that it shouldn't - while on the other hand, if you use either `src` or `.` then you'd end up excluding things that should be excluded. I do not believe that we are using in this in localized repos, but if we are you could also add logic for that too. I'll let you decide -- just let me know.

Reply:
What I mean is not to exclude everything, and also not to look for `.` or `repo-name`. Instead, find which `src` pattern matches a file on disk, there should be exactly one match.

Reply:
@IEvangelist Actually, [this repo](https://github.com/MicrosoftDocs/live-share/blob/master/docs/docfx.json) doesn't have a "src" specified at all, which may default `.`, but it has excludes.

Reply:
Yes, if there is no `src` assume it's `"."` and use those `excludes`.

Reply:
@IEvangelist I think it's a bit more complicated. The `docfx.json` *may* not be in the root, which is an incorrect assumption I have, and there may be multiple `docfx.json` files. I'm going to look into fixing that.

Reply:
I agree, and you're not wrong -- but I just think that for the scope of this specific enhancement we're trying to boil the ocean. While there are lots of additional considerations and edge cases, the reality is that as of right now and for the immediate future, this will only be used by a select few repos. IMO, it's better to prioritize around standardization, and create issues for future consideration that capture these potential concerns.

As of now, we're reviewing this current PR - and I think will it's ok it could be improved by what I was suggesting. Is that fair?

Reply:
@IEvangelist I think I could be near to get the more generalized approach working, but happy to switch back to the simpler approach, which as far as I understood works with like this:

- Always assume a single `docfx.json` in repo root.
- Exclude `content` that has `src` == `.` or `repo-name` (assuming the GH Action isn't run in the localized mirrors).